server/pkg/api/accountHandlers.go

```diff
@@ -6,16 +6,78 @@ import (
 
 	"github.com/darwinfroese/hacksite/server/models"
 	"github.com/darwinfroese/hacksite/server/pkg/accounts"
+	"github.com/darwinfroese/hacksite/server/pkg/auth"
 )
 
-var accountHandlerMap = map[string]handler{
+var accountsHandlerMap = map[string]handler{
 	"POST": createAccount,
 }
 
+var accountHandlerMap = map[string]handler{
+	"GET": getAccount,
+	"POST": updateAccount,
+}
+
 func (ctx *Context) accountsRoute(w http.ResponseWriter, r *http.Request) {
+	callHandler(ctx, w, r, accountsHandlerMap)
+}
+
+func (ctx *Context) accountRoute(w http.ResponseWriter, r *http.Request) {
 	callHandler(ctx, w, r, accountHandlerMap)
 }
 
+func getAccount(ctx *Context, w http.ResponseWriter, r *http.Request) {
+	var account models.Account
+
+	session, err := auth.GetCurrentSession(*ctx.DB, *ctx.Logger, r)
+	if err != nil {
+		(*ctx.Logger).ErrorWithRequest(r, ctx.RequestID, err.Error())
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+
+	account, err = (*ctx.DB).GetAccountByUsername(session.Username)
+	if err != nil {
+		(*ctx.Logger).ErrorWithRequest(r, ctx.RequestID, err.Error())
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	json.NewEncoder(w).Encode(account)
+}
+
+func updateAccount(ctx *Context, w http.ResponseWriter, r *http.Request) {
+	var updateData models.Account
+	var account models.Account
+
+	err := json.NewDecoder(r.Body).Decode(&updateData)
+
+	if err != nil {
+		(*ctx.Logger).ErrorWithRequest(r, ctx.RequestID, err.Error())
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	session, err := auth.GetCurrentSession(*ctx.DB, *ctx.Logger, r)
+	if err != nil {
+		(*ctx.Logger).ErrorWithRequest(r, ctx.RequestID, err.Error())
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+
+	account, err = (*ctx.DB).GetAccountByUsername(session.Username)
+	if err != nil {
+		(*ctx.Logger).ErrorWithRequest(r, ctx.RequestID, err.Error())
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	account.Password = updateData.Password
+
+	json.NewEncoder(w).Encode(account)
```

Comment:
we should probably also `w.WriteHeader` 200 on get and 204 without the body on update.